packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1496,92 +1543,271 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
   }
 }
 
+/// An edge of a box, such as top, bottom, left or right, used to compute
+/// [SemanticsNode]s that overlap vertically or horizontally.
```

Comment:
Maybe document: For computing horizontal overlap in an LTR setting we create two _BoxEdge objects for each Semantics Node: one representing the left edge (marked with isStart: true) and one for the right edge (with isStart: false). Similarly, for vertical overlap you also create two objects for each SemanticsNode, one for the top and one for the bottom edge.

Reply:
Done.